File: python/chol_factor_test.py
import numpy as np
from chol_factor import chol_factor
from triangular import triangular

# TEST: Cholesky factorization (LL')

# Symmetric positive definite matrix
A = np.matrix('5    1.2  0.3 -0.6;'
              '1.2  6   -0.4  0.9;'
              '0.3 -0.4  8    1.7;'
             '-0.6  0.9  1.7  10');
print('A = \n', A)
# Computation of the L factor
L = chol_factor(A)
print('L = \n', L)
# Check
if np.allclose(A, np.dot(L, L.transpose())) == False:
    raise Exception('QR factorizzation test failure')


# TEST: System Resolution

# Ax = LL'x = b
b = np.matrix("68; 9; 45; 35")
print('b = \n', b)
# Lk = b
k = triangular(L, b, 1)
print('k = \n', k)
# L'x = k
x = triangular(L.transpose(), k, 0)
print('x = \n', x)
# Check
b1 = np.dot(A, x)
print('b1 = \n', b1)
if np.allclose(b, b1) == False:
    raise Exception('System resolution failure')

File: octave/chol_factor_test.m
% TEST: Cholesky factorization (LL')

% Symmetric positive definite matrix
A = [ 5    1.2  0.3 -0.6;
      1.2  6   -0.4  0.9;
      0.3 -0.4  8    1.7;
     -0.6  0.9  1.7  10 ]
% Computation of the L factor
L = chol_factor(A)
% Check
if allclose(A, L*L', 0.0001) == 0
    error('Cholesky factorization test failure')
end

% TEST: System Resolution

% Ax = LL'x = b
b = [ 68; 9; 45; 35 ]
% Lk = b
k = triangular(L, b, 1)
% L'x = k
x = triangular(L', k, 0)
% Check
b1 = A*x
if allclose(b, b1, 0.0001) == 0
    error('System resolution failure')
end
